### DISCUSSION TITLE
IDE 2.3 + rp2040: can't build on macos 11

### DISCUSSION THREAD

**Discussion by myklemykle** [earlephilhower/arduino-pico#1993] [Q&A]
Hi, I just hit an Arduino config bug, involving the version of python3 in rp2040/tools .  I'm really not sure if it's a thing to be addressed in this core or in the main Arduino system.  [I reported it over there](https://forum.arduino.cc/t/ide-2-3-rp2040-cant-build-on-macos-11/1221975/1), but I want to ask here as well.

Basically, a version of python3 gets installed under rp2040/tools/ that's built for macos 14.  Arduino 2.3 claims to support macos versions down to 11, but this python3 binary does not run on that os version.

I'm not sure if arduino-pico even has a theory of macos version support, or if it needs one. But in this case I don't know what part of the Arduino multiverse is responsible for choosing this version of python3 . Are the contents of rp2040/tools specified by the arduino-pico core, or filled in by Arduino?

**Comment by earlephilhower** [comment]
Sorry, that has more to do with what Apple deigns to support and what GitHub Apple M1 VMs allow for SDK installations.

These are the XCode versions supported by GH's M1 VMs, which are the only way I can build Python3 for Apple.  I chose the [lowest version they allow](https://github.com/earlephilhower/pico-quick-toolchain/actions/runs/7729850389/job/21073949406).  Note that the XCode version != the MacOS version, so 14.3.1 corresponds to [MacOS 13 support](https://xcodereleases.com/):

```` 
┌─────────┬──────────┬─────────────┬─────────────┬────────┬──────────────────────────────────┐
│ (index) │ version  │ buildNumber │ releaseType │ stable │               path               │
├─────────┼──────────┼─────────────┼─────────────┼────────┼──────────────────────────────────┤
│    0    │ '15.2.0' │  '15C500b'  │    'GM'     │  true  │  '/Applications/Xcode_15.2.app'  │
│    1    │ '15.1.0' │   '15C65'   │    'GM'     │  true  │  '/Applications/Xcode_15.1.app'  │
│    2    │ '15.0.1' │  '15A507'   │    'GM'     │  true  │ '/Applications/Xcode_15.0.1.app' │
│    3    │ '14.3.1' │  '14E300c'  │    'GM'     │  true  │ '/Applications/Xcode_14.3.1.app' │
└─────────┴──────────┴─────────────┴─────────────┴────────┴──────────────────────────────────┘
````

That said, MacOS 11 is EOL and unsupported by even Apple, [per Wackypedia](https://en.wikipedia.org/wiki/MacOS).  So I would have to suggest moving to something newer on your machine, if only for security updates/etc.
